### ISSUE TITLE
[java] MethodArgumentCouldBeFinal shouldn't report unused parameters

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#4512]
**Affects PMD Version:** 6.x

**Rule:** [MethodArgumentCouldBeFinal](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#methodargumentcouldbefinal)

**Description:**

Similar like #4511: Unused formal parameters shouldn't be reported. This is a different problem [UnusedFormalParameter](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedformalparameter)

This has been found via #3123. It is already fixed with PMD 7.0.0-rc1.
Has been fixed with #3113.

**Code Sample demonstrating the issue:**

```java
public class Foo {
    public static void main(String[] args) {}
}
```

**Expected outcome:**

PMD reports a violation at line 2, but that's wrong. That's a false positive.
